pkg/cloud/services/eks/nodegroup.go

```diff
@@ -337,17 +338,38 @@ func (s *NodegroupService) reconcileNodegroupVersion(ng *eks.Nodegroup) error {
 			updateMsg = fmt.Sprintf("to AMI version %s", *input.ReleaseVersion)
 		}
 
+		conditions.MarkFalse(
+			s.scope.ManagedMachinePool,
+			expinfrav1.EKSNodegroupUpdateSucceededCondition,
+			expinfrav1.EKSNodegroupUpdatingReason,
+			clusterv1.ConditionSeverityInfo,
+			"",
+		)
 		if err := wait.WaitForWithRetryable(wait.NewBackoff(), func() (bool, error) {
 			if _, err := s.EKSClient.UpdateNodegroupVersion(input); err != nil {
+				conditions.MarkFalse(
```

Comment:
We probably don't need to set the condition here as it is covered by the error handling block 364-384

Reply:
Updated